support-frontend/assets/helpers/useHasBeenSeen.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+// $FlowIgnore - required for hooks
+import { useEffect, useState, useRef } from 'react';
+import libDebounce from 'lodash.debounce';
+
+const useHasBeenSeen = (options: IntersectionObserverOptions & { debounce?: boolean }) => {
+  const [hasBeenSeen, setHasBeenSeen] = useState<boolean>(false);
+  const [node, setNode] = useState<HTMLElement | null>(null);
+
+  const observer = useRef<IntersectionObserver | null>(null);
+
+  // Enabling debouncing ensures the target element intersects for at least
+  // 200ms before the callback is executed
+  const intersectionFn: IntersectionObserverCallback = ([entry]) => {
+    if (entry.isIntersecting) {
+      setHasBeenSeen(true);
+    }
+  };
+  const intersectionCallback = options.debounce
+    ? libDebounce(intersectionFn, 200)
+    : intersectionFn;
+
+  useEffect(() => {
+    if (observer.current) {
+      observer.current.disconnect();
+    }
+
+    observer.current = new window.IntersectionObserver(
```

Comment:
Are we polyfilling IntersectionObserver, or will this otherwise fail gracefully in IE?

Reply:
Good shout. Looks like we're using a different polyfill.io path on support.theguardian.com compared to the main domain, ours unfortunately doesn't polyfill, whereas there's does. It should be resolvable by adding "IntersectionObserver. IntersectionObserverEntry" to the URL.

Reply:
OK I've updated the polyfill script we use to match the script used on [dotcom-rendering](https://github.com/guardian/dotcom-rendering/blob/82c6327083272d2ec8f563e15706ac6a933b8224/src/web/server/document.tsx#L173) and `frontend`. This script lives on `assets.guim.co.uk`, so required a change to our Content Security Policy to enable it. This script includes the IntersectionObserver polyfill.

I've tested before/after on IE11 using Browserstack, and the new Polyfill works as expected.

I noted whilst doing this we had a random 50/50 test for loading v2/v3 of the polyfill.io script, this was set-up back in June 2019: https://github.com/guardian/support-frontend/pull/1872. I guess it's been in place this entire time. I've decided to remove it as I don't think we're still monitoring the results?

Pinging @guardian/contributions for visibility on this change.